diff --git a/dist/async_hooks/index.mjs b/dist/async_hooks/index.mjs
--- a/dist/async_hooks/index.mjs
+++ b/dist/async_hooks/index.mjs
@@ -1,5 +1,6 @@
 //#region src/async_hooks/index.ts
-const AsyncLocalStoragePromise = import(
+const globalAsyncLocalStorage = globalThis.AsyncLocalStorage;
+const AsyncLocalStoragePromise = globalAsyncLocalStorage ? Promise.resolve(globalAsyncLocalStorage) : import(
 	/* @vite-ignore */
 	/* webpackIgnore: true */
 	"node:async_hooks"
